VeronicaM/residence
# feat: API

# API

## What is this change ? 

- Integrate json-server
- Add GET and POST functionality to fetch and create residence data
- Render Residence Form in main app
- Run the server and client concurrently to start the app in a single step
- Setup circle ci to run tests
- Add lint, format and test coverage npm scripts

## How to test 
- Run `yarn start`
- Check that the Residence form is displayed correctly on page load
- Run `yarn test` and check that all tests pass
- Run `yarn lint` and check that there are no lint errors 
- Check that you can add a new residence through the form
- Check that residence data is fetched on page load by inspecting the network tab to see the GET request for residences being made
